Rules/CompatibilityRules/CompatibilityRule.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Management.Automation.Language;
+using System.Text.RegularExpressions;
+using Microsoft.PowerShell.CrossCompatibility.Query;
+using Microsoft.PowerShell.CrossCompatibility.Utility;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+    public abstract class CompatibilityRule : ConfigurableRule
+    {
+        private const string PROFILE_DIR_NAME = "compatibility_profiles";
```

Comment:
I gave the profile dir here a different (more explicit) name than in the CrossCompatibility module. It might be better off as a static readonly so that we can use the same variable everywhere...